components/kf-monitoring/MetricsExporter.go

```diff
@@ -0,0 +1,161 @@
+package monitoring_util
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/log"
+	"net"
+	"net/http"
+)
+
+// Common label keys for all metrics signals
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+// COMPONENT: name of Component outputing the metrics, eg. "tf-operator"
+const COMPONENT = "Component"
+// KIND: each componenet can label their metrics with custom tag "kind". Suggest keeping "kind" value to be CONSTANT PER METRICS.
+const KIND = "kind"
+const NAMESPACE = "namespace"
+// ACTION: request action type of the metrics, eg. "CRUD"
+const ACTION = "action"
+// SEVERITY: level of importance, used to filter alerts
+const SEVERITY  = "severity"
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+// Alerting metrics severity levels
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+const SEVERITY_MINOR = "Minor"
+const SEVERITY_MAJOR = "Major"
+const SEVERITY_CRITICAL = "Critical"
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+// Util const values
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+const METRICSPORT = 8079
+const METRICSPATH = "/metrics"
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+var (
+	// Counter metrics
+	// num of requests counter vec
+	requestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "request_counter",
+			Help: "Number of request_counter",
+		},
+		[]string{COMPONENT, KIND, NAMESPACE, ACTION, SEVERITY},
+	)
+	// Counter metrics for failed requests
+	requestFailureCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "request_failure_counter",
+		Help: "Number of request_failure_counter",
+	}, []string{COMPONENT, KIND, NAMESPACE, ACTION, SEVERITY})
+
+	// Gauge metrics
+	requestGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "requests_gauge",
```

Comment:
Requests should not be a gauge value. From my thought, requests are only appropriate for counter.